lib/spack/spack/util/compression.py

```diff
@@ -649,3 +745,37 @@ def check_and_remove_ext(path, ext):
     if check_extension(path, ext):
         return reg_remove_ext(path, ext)
     return path
+
+
+def reg_sub_ext(path, old_ext, new_ext):
+    return re.sub(rf"{old_ext}", rf"{new_ext}", path)
+
+
+def expand_contracted_extension_in_path(path, ext=None):
+    if not ext:
+        ext = extension_from_path(path)
+    expanded_ext = expand_contracted_extension(ext)
+    if expanded_ext != ext:
+        return reg_sub_ext(path, ext, expanded_ext)
+    return path
+
+
+def expand_contracted_extension(extension):
+    if extension in NOTAR_EXTS and extension != "zip":
```

Comment:
No, extension is almost never preceded by a `.`, and the only way it could be is if the caller explicitly made this call with an extension prefixed with `.` (from `decompressor_for` or by calling this directly), but all entrypoints from this module would guarantee there's no `.` in front of the prefix. Will add the `.` stripping.